compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/AMDAIEDmaUtilsTest.cpp

```diff
@@ -271,6 +271,97 @@ TEST_F(AccessPatternCombinationTest, FailCombineAccessPatterns) {
                              {128, 1}, {32, 0}, {96, 64}, {128, 1}, 4, false);
 }
 
+class FoldLinearDimsTest : public ::testing::Test {
+ protected:
+  FoldLinearDimsTest() : rewriter(&context), loc(UnknownLoc::get(&context)) {
+    context.loadDialect<arith::ArithDialect>();
+  }
+
+  SmallVector<OpFoldResult> toOpFoldResult(const SmallVector<int64_t> &values) {
+    return llvm::map_to_vector(values, [&](int64_t v) -> OpFoldResult {
+      return getAsIndexOpFoldResult(&context,
+                                    v);  // rewriter.getI64IntegerAttr(v);
+    });
+  }
+
+  void checkFoldLinearDims(const SmallVector<int64_t> offsets,
+                           const SmallVector<int64_t> sizes,
+                           const SmallVector<int64_t> strides,
+                           ArrayRef<int64_t> maxSizes,
+                           const SmallVector<int64_t> expectedOffsets,
+                           const SmallVector<int64_t> expectedSizes,
+                           const SmallVector<int64_t> expectedStrides,
+                           bool shouldSucceed = true) {
+    SmallVector<OpFoldResult> offsetsValues = toOpFoldResult(offsets);
+    SmallVector<OpFoldResult> sizesValues = toOpFoldResult(sizes);
+    SmallVector<OpFoldResult> stridesValues = toOpFoldResult(strides);
+    SmallVector<OpFoldResult> expectedOffsetsValues =
+        toOpFoldResult(expectedOffsets);
+    SmallVector<OpFoldResult> expectedSizesValues =
+        toOpFoldResult(expectedSizes);
+    SmallVector<OpFoldResult> expectedStridesValues =
+        toOpFoldResult(expectedStrides);
+    SmallVector<OpFoldResult> newOffsets;
+    SmallVector<OpFoldResult> newSizes;
+    SmallVector<OpFoldResult> newStrides;
+    if (shouldSucceed) {
+      EXPECT_TRUE(succeeded(foldLinearDims(&context, offsetsValues, sizesValues,
+                                           stridesValues, newOffsets, newSizes,
+                                           newStrides, maxSizes)));
+      EXPECT_EQ(newOffsets, expectedOffsetsValues);
```

Comment:
Common trailing code in if- and else- can be moved after.

Reply:
Thanks for the catch, updated it.